src/testdata/ledger-close-meta-v1-protocol-20.json

```diff
@@ -6,24 +6,24 @@
                 "v": 0
             },
             "ledgerHeader": {
-                "hash": "a50c06f398679584d4acff68de5130502114df913f41c270a4bce9e1d6cd5c36",
+                "hash": "96bbf541d371216daf6d52718c6c80c40cbfa67b51e578df821c238af37fcf6b",
```

Comment:
Do we understand the reason for the LCM changes?

Reply:
If I understand correctly, when we manually do soroban network config upgrades, we call `LiveBucketList::addBatch` twice in the same ledger, so we have two different snapshots with the same ledgerSeq. LedgerSeq is what we use to determine if a snapshot is out of date previously in `maybeUpdateSnapshot`, so when doing manual upgrades, the child snapshot would be stale (the child snapshot is based off of the 1st addBatch call, not the 2nd), but would not be updated since the ledgerSeq of the stale snapshot is still "up to date". This PR changes is such that `maybeUpdateSnapshot` just checks for a null pointer but doesn't actually check the ledgerSeq, so child snapshots are always updated when they call `maybeUpdateSnapshot`.

Reply:
The main reason the meta test broke was because I removed snapshot auto-updates, and reset the snapshot for each new LedgerTxn (previously it had a long-living pointer that called into maybeUpdateSnapshot). So because of the reset, LedgerTxn now always picks up the most up-to-date snapshot (in this case, the one from the 2nd call to addBatch). This was not the case previously, as that test used the first snapshot. Note that the meta breaks regardless of the ledgerSeq check in `maybeUpdateSnapshot` (that change followed later).